src/util/linear_scan.rs

```diff
@@ -0,0 +1,75 @@
+use crate::util::alloc_bit;
+use crate::util::Address;
+use crate::util::ObjectReference;
+use crate::vm::ObjectModel;
+use crate::vm::VMBinding;
+use std::marker::PhantomData;
+
+/// Iterate over an address range, and find each object by alloc bit.
+/// ATOMIC_LOAD_ALLOC_BIT can be set to false if it is known that loading alloc bit
+/// non-atomically is correct (e.g. a single thread is scanning this address range, and
+/// it is the only thread that accesses alloc bit).
+pub struct ObjectIterator<VM: VMBinding, S: LinearScanObjectSize, const ATOMIC_LOAD_ALLOC_BIT: bool>
```

Comment:
I was wondering if the type parameter `LinearScanObjectSize` is required? Would a function pointer to a `size()` function not be good enough? Or are you trying to reduce overhead of indirect function calls? (Is it even significant enough?). Either way, it's not a big deal, but was wondering about your thoughts.

Reply:
Yeah, I was considering about performance and trying to be safe, as we know using trait would have no overhead at all. Maybe using a function pointer is enough.